src/main/scala/chisel3/experimental/inlinetest/InlineTest.scala

```diff
@@ -88,39 +90,49 @@ object TestHarnessGenerator {
   }
 }
 
+private class TestGenerator[M <: RawModule, R](
+  dutName:              String,
+  testName:             String,
+  testBody:             Instance[M] => R,
+  resetType:            Option[Module.ResetType.Type],
+  testHarnessGenerator: TestHarnessGenerator[M, R]
+) {
+  def generate(dutDefinition: Definition[M]) =
+    testHarnessGenerator.generate(new TestParameters(dutName, testName, dutDefinition, testBody, resetType))
+}
+
 /** Provides methods to build unit testharnesses inline after this module is elaborated.
   *
   *  @tparam TestResult the type returned from each test body generator, typically
   *  hardware indicating completion and/or exit code to the testharness.
   */
 trait HasTests[M <: RawModule] { module: M =>
 
-  /** A Definition of the DUT to be used for each of the tests. */
-  private lazy val moduleDefinition =
-    module.toDefinition.asInstanceOf[Definition[module.type]]
-
-  /** Generate an additional parent around this module.
-    *
-    *  @param parent generator function, should instantiate the [[Definition]]
-    */
-  protected final def elaborateParentModule(parent: Definition[module.type] => RawModule with Public): Unit =
-    afterModuleBuilt { Definition(parent(moduleDefinition)) }
+  /** Generators for inline tests by name. */
+  private val testGenerators = new mutable.HashMap[String, (Int, TestGenerator[M, _])]
 
   /** Generate a public module that instantiates this module. The default
     *  testharness has clock and synchronous reset IOs and contains the test
     *  body.
     *
-    *  @param testBody the circuit to elaborate inside the testharness
+    *  @param testBody the circuit to generate inside the testharness
```

Comment:
```suggestion
    *  @param testBody the circuit to elaborate inside the testharness
```

Unintentional change